diff --git a/src/model/diffusion/unet.hpp b/src/model/diffusion/unet.hpp
index 1b0e453..16c3cf7 100644
--- a/src/model/diffusion/unet.hpp
+++ b/src/model/diffusion/unet.hpp
@@ -197,8 +197,8 @@ public:
         // GGML_ASSERT(ggml_n_dims(context) == 3);
 
         auto norm             = std::dynamic_pointer_cast<GroupNorm32>(blocks["norm"]);
-        auto proj_in          = std::dynamic_pointer_cast<Conv2d>(blocks["proj_in"]);
-        auto proj_out         = std::dynamic_pointer_cast<Conv2d>(blocks["proj_out"]);
+        auto proj_in          = std::dynamic_pointer_cast<UnaryBlock>(blocks["proj_in"]);
+        auto proj_out         = std::dynamic_pointer_cast<UnaryBlock>(blocks["proj_out"]);
         auto time_pos_embed_0 = std::dynamic_pointer_cast<Linear>(blocks["time_pos_embed.0"]);
         auto time_pos_embed_2 = std::dynamic_pointer_cast<Linear>(blocks["time_pos_embed.2"]);
         auto time_mixer       = std::dynamic_pointer_cast<AlphaBlender>(blocks["time_mixer"]);
@@ -228,10 +228,15 @@ public:
         time_context                     = ggml_repeat(ctx->ggml_ctx, time_context_first_timestep, time_context);  // [b*h*w, n_context, context_dim]
 
         x = norm->forward(ctx, x);
-        x = proj_in->forward(ctx, x);  // [N, inner_dim, h, w]
-
-        x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 2, 0, 3));  // [N, h, w, inner_dim]
-        x = ggml_reshape_3d(ctx->ggml_ctx, x, inner_dim, w * h, n);                // [N, h * w, inner_dim]
+        if (use_linear) {
+            x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 2, 0, 3));
+            x = ggml_reshape_3d(ctx->ggml_ctx, x, inner_dim, w * h, n);
+            x = proj_in->forward(ctx, x);
+        } else {
+            x = proj_in->forward(ctx, x);
+            x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 2, 0, 3));
+            x = ggml_reshape_3d(ctx->ggml_ctx, x, inner_dim, w * h, n);
+        }
 
         auto num_frames = ggml_arange(ctx->ggml_ctx, 0.f, static_cast<float>(timesteps), 1.f);
         // since b is 1, no need to do repeat
@@ -275,12 +280,17 @@ public:
             x = time_mixer->forward(ctx, x, x_mix);  // [N, h * w, inner_dim]
         }
 
-        x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 0, 2, 3));  // [N, inner_dim, h * w]
-        x = ggml_reshape_4d(ctx->ggml_ctx, x, w, h, inner_dim, n);                 // [N, inner_dim, h, w]
-
-        // proj_out
-        x = proj_out->forward(ctx, x);  // [N, in_channels, h, w]
+        if (use_linear) {
+            x = proj_out->forward(ctx, x);
+            x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 0, 2, 3));
+            x = ggml_reshape_4d(ctx->ggml_ctx, x, w, h, inner_dim, n);
+        } else {
+            x = ggml_cont(ctx->ggml_ctx, ggml_permute(ctx->ggml_ctx, x, 1, 0, 2, 3));
+            x = ggml_reshape_4d(ctx->ggml_ctx, x, w, h, inner_dim, n);
+            x = proj_out->forward(ctx, x);
+        }
 
         x = ggml_add(ctx->ggml_ctx, x, x_in);
         return x;
